src/main/java/com/commercetools/sync/shoppinglists/utils/ShoppingListReferenceResolutionUtils.java

```diff
@@ -177,56 +188,49 @@ private static LineItemDraft mapToLineItemDraft(@Nonnull final LineItem lineItem
+      @Nullable final List<TextLineItem> textLineItems,
+      @Nonnull final Map<String, String> referenceIdToKeyMap) {
 
     if (textLineItems == null) {
       return null;
     }
 
     return textLineItems.stream()
         .filter(Objects::nonNull)
-        .map(ShoppingListReferenceResolutionUtils::mapToTextLineItemDraft)
+        .map(textLineItem -> mapToTextLineItemDraft(textLineItem, referenceIdToKeyMap))
         .collect(toList());
   }
 
   @Nonnull
   private static TextLineItemDraft mapToTextLineItemDraft(
-      @Nonnull final TextLineItem textLineItem) {
+      @Nonnull final TextLineItem textLineItem,
+      @Nonnull final Map<String, String> referenceIdToKeyMap) {
 
     return TextLineItemDraftBuilder.of(textLineItem.getName(), textLineItem.getQuantity())
         .description(textLineItem.getDescription())
         .addedAt(textLineItem.getAddedAt())
-        .custom(mapToCustomFieldsDraft(textLineItem.getCustom()))
+        .custom(mapToCustomFieldsDraft(textLineItem.getCustom(), referenceIdToKeyMap))
         .build();
   }
 
   /**
    * Builds a {@link ShoppingListQuery} for fetching shopping lists from a source CTP project with
-   * all the needed references expanded for the sync:
+   * only the Variants expanded for the sync:
    *
    * <ul>
-   *   <li>Customer
-   *   <li>Custom Type of the Shopping List
    *   <li>Variants of the LineItems
-   *   <li>Custom Types of the LineItems
-   *   <li>Custom Types of the TextLineItems
    * </ul>
    *
-   * <p>Note: Please only use this util if you desire to sync all the aforementioned references from
-   * a source commercetools project. Otherwise, it is more efficient to build the query without
-   * expansions, if they are not needed, to avoid unnecessarily bigger payloads fetched from the
-   * source project.
+   * <p>Note: Please only use this util if you desire to sync the aforementioned reference from a
+   * source commercetools project. Otherwise, it is more efficient to build the query without
+   * expansion, if it is not needed, to avoid unnecessarily bigger payloads fetched from the source
+   * project.
    *
-   * @return the query for fetching shopping lists from the source CTP project with all the
-   *     aforementioned references expanded.
+   * @return the query for fetching shopping lists from the source CTP project without references
+   *     expanded except for Variants of the LineItems.
    */
   public static ShoppingListQuery buildShoppingListQuery() {
-    return ShoppingListQuery.of()
-        .withExpansionPaths(ShoppingListExpansionModel::customer)
-        .plusExpansionPaths(ExpansionPath.of("custom.type"))
-        .plusExpansionPaths(ExpansionPath.of("lineItems[*].variant"))
-        .plusExpansionPaths(ExpansionPath.of("lineItems[*].custom.type"))
-        .plusExpansionPaths(ExpansionPath.of("textLineItems[*].custom.type"));
+    return ShoppingListQuery.of().plusExpansionPaths(ExpansionPath.of("lineItems[*].variant"));
```

Comment:
Yes, Because for variant's we are fetching SKU and quantity values but not the key values. And In service class also we are using Variant expansion in this case. So, discussed this already with @ahmetoz and I left variant to expand in the query for now.